pymongo/asynchronous/client_session.py

```diff
@@ -693,7 +699,10 @@ async def callback(session, custom_arg, custom_kwarg=None):
                 read_concern, write_concern, read_preference, max_commit_time_ms
             )
             try:
-                ret = callback(self)
+                if not _IS_SYNC and iscoroutinefunction(callback):
```

Comment:
Why check IS_SYNC or iscoroutinefunction here? Shouldn't it just be:
```python
ret = await callback(self)
```

Reply:
I was trying to be more lenient and allow both async and sync callbacks, but it's better to require all asynchronous callbacks for the Async API.